src/site/content/en/blog/smoothness/index.md

```diff
@@ -0,0 +1,563 @@
+  API or the Event Timing API. Yet, there are also features, such as
+  [`isInputPending()`](/isinputpending/), where long tasks during idle periods
+  may be added entirely by design and be a good thing.
+{% endAside %}
+
+
+## Animation frames: Updates that matter
+
+The above example showcases that there is more to the story than just
+`requestAnimationFrame()`.
+
+So when do animation updates, and animation frames, actually matter?  Here are
+some criteria that we think about, and we'd love to get feedback on:
+
+- Main and compositor thread updates
+- Missing paint updates
+- Detecting animations
+- Quality versus quantity
+
+### Main and compositor thread updates
+
+First of all, animation frame updates are not boolean. It is not the case that
+frames may only be fully dropped or fully presented. There are many reasons why
+an animation frame may be _partially_ _presented_. In other words, it can
+simultaneously have _some stale content_ while also having _some new visual
+updates_ which are presented.
+
+The most common example of this is when the browser is unable to produce a new main
+thread update within frame deadline but does have a new compositor thread update
+(such as the threaded scrolling example from earlier).
+
+One important reason why using declarative animations to animate composite
+properties is so recommended, is precisely because that enables an animation to
+be driven entirely by the compositor thread even when the main thread is blocked
+processing tasks! These types of animations can continue to produce visual
+updates efficiently and in parallel.
+
+On the other hand, there may be cases where a main thread update finally becomes
+available for presentation, but only after missing several frame deadlines. Here
+the browser will have _some_ new update, but it may not be _the very latest_.
+
+Broadly speaking, we consider frames that contain some new visual updates,
+without all new visual updates, as a _partial frame_. Partial frames are fairly
+common. Ideally, partial updates would at least include the most important
+visual updates, like animations, but that can only happen if animations are driven by the compositor thread.
+
+### Missing paint updates
+
+Another type of partial update is when media like images are simply not done
+painting (decoding and rasterizing) in time for frame presentation.
+
+Or, even if a page is perfectly static, browsers may still fall behind rendering
+visual updates during rapid scrolling. That is because the pixel renditions of
+content beyond the visible viewport may be discarded to save GPU memory. It
+takes time to render pixels, and it may take longer than a single frame to
+render everything after a large scroll, like a finger fling. This is commonly
+known as _checkerboarding_.
+
+With each frame rendering opportunity, it's possible to track how much of the
+latest visual updates actually got to the screen. This is broadly known as
+_frame throughput_.
```

Comment:
I think this line needs updating to something like: 

"With each frame rendering opportunity, it's possible to track how much of the
latest visual updates actually got to the screen.  Measuring the ability to do so over many frames (or time) is broadly known as _frame throughput_."

...in other words, throughput isn't something you do for a single frame, its for a span of time.  The existing text may not make that clear.

Reply:
Updated.